Sources/PackagePlugin/PackageManagerProxy.swift

```diff
@@ -29,14 +29,14 @@ public struct PackageManager {
     ) throws -> BuildResult {
         // Ask the plugin host to build the specified products and targets, and wait for a response.
         // FIXME: We'll want to make this asynchronous when there is back deployment support for it.
-        return try sendMessageAndWaitForReply(.buildOperationRequest(subset: subset, parameters: parameters)) {
+        return try sendMessageAndWaitForReply(.buildOperationRequest(subset: .init(subset), parameters: .init(parameters))) {
             guard case .buildOperationResponse(let result) = $0 else { return nil }
-            return result
+            return .init(result)
         }
     }
     
     /// Specifies a subset of products and targets of a package to build.
-    public enum BuildSubset: Encodable {
```

Comment:
These structs, which are part of the API, were all directly serialized to JSON.  The evolution proposal doesn't state that they should be Codable, and this was a case of implementation leaking through.  Now the serialized structure is separate from the API.

Reply:
that is a great change, the model and API should be separate from the serialization format as it buys us an important degree of freedom (even if at the cost of some healthy duplication)

Reply:
I thought so too.  I'm wondering if there's a way to automate some of that duplication, e.g. with .gyb or something.